server-spi-private/src/main/java/org/keycloak/services/clientpolicy/executor/ClientPolicyExecutorProvider.java

```diff
@@ -34,6 +34,14 @@
     default void close() {
     }
 
+    /**
+     * setup this executor's configuration.
+     *
+     * @param config
+     */
+    default void setupConfiguration(Object config) {
```

Comment:
I've tried to do some simplification in my branch https://github.com/mposolda/keycloak/tree/KEYCLOAK-16805-client-policies-db-change (See the commit https://github.com/mposolda/keycloak/commit/783e9c3a2d130eb6e1a3bd92c2b1679982af02f2 ) . I've tried to add some common configuration class, which will be common to all the ClientPolicyExecutorProvider. At the same time, I've tried to add some type-safety and also wanted to avoid some boilerplate code with parsing the JSON configuration to be in each executor. After I did this, I figured that for ClientPolicyConditionProvider, the situation is better than for the executors and you already added some common class (AbstractClientCondition and the common configuration class AbstractClientCondition.Configuration).

I suggest that both ClientPolicyExecutorProvider and ClientPolicyConditionProvider use some approach with the "common" configuration class and type-safety. And that both ClientPolicyExecutorProvider and ClientPolicyConditionProvider are aligned and use similar approach. I suggest to either:
1) Cherry-pick my commit https://github.com/mposolda/keycloak/commit/783e9c3a2d130eb6e1a3bd92c2b1679982af02f2 to this PR and refactor the ClientPolicyConditionProvider to use same approach aligned with the ClientPolicyExecutorProvider I introduced.
2) Change executors a bit to align with the approach you used for the conditions (AbstractClientCondition and AbstractClientCondition.Configuration class)

Do you prefer (1) or (2) or something else? WDYT?

Reply:
@mposolda I would like to choose the option (1) you've proposed.

Reply:
@tnorimat Thanks, will wait for your updates on this PR.